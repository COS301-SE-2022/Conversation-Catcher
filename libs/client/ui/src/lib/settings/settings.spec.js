import React from 'react';
import { render } from '@testing-library/react-native';
import Settings from './settings';
import { create } from 'react-test-renderer';
import { configureStore } from '@reduxjs/toolkit';
import reducer from '../../../../../../apps/client/src/app/slices/colour.slice';
import { Provider } from 'react-redux';

describe('Setup of settings', () => {
  const store = configureStore({
    reducer: {
        colour:reducer
    }
  })
  const TestComponent = () => (
    <Provider store = { store }>
        <Settings />
    </Provider>
)
  it('should render successfully', () => {
    const { container } = render(<TestComponent />);
    expect(container).toBeTruthy();
  });


  // const tree = create(<Settings />);
  // test('snapshot', () => {
  //   expect(tree).toMatchSnapshot();
  // });

});

// describe('The actions that occur on button press', () => {
//   it('should display the available colours')
// });
