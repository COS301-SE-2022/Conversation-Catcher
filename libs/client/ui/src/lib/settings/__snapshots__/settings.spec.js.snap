// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setup of settings snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffffff",
        "overflow": "hidden",
      },
      Object {
        "flexGrow": 1,
        "flexShrink": 0,
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
        "minHeight": 844,
      },
    ]
  }
>
  <View
    style={
      Object {
        "flexGrow": 1,
      }
    }
    x="12.5px 365fr 12.5px"
    y="60px minmax(0px, max-content) 60px"
  >
    <View
      style={
        Object {
          "flexGrow": 0,
          "flexShrink": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexDirection": "row",
            "flexGrow": 1,
            "justifyContent": "center",
          }
        }
        x="0% 34.52% 65.48%"
        y="51px minmax(0px, max-content) 0px"
      >
        <Text
          ellipsizeMode="clip"
          style={
            Object {
              "color": "#344053ff",
              "fontFamily": "System",
              "fontSize": 30,
              "fontStyle": "normal",
              "fontWeight": "700",
              "letterSpacing": 0,
              "lineHeight": 30,
              "paddingHorizontal": 0,
              "paddingVertical": 0,
              "textAlign": "center",
            }
          }
        >
          Settings
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexGrow": 0,
          "flexShrink": 1,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#f5f5f5ff",
            "borderRadius": 7,
            "flexGrow": 1,
            "width": "100%",
          }
        }
        x="12.5px 340fr 12.5px"
        y="22px minmax(0px, max-content) 0px"
      >
        <View
          style={
            Object {
              "flexGrow": 1,
            }
          }
          x="24px 292fr 24px"
          y="0px minmax(0px, max-content) 0px"
        >
          <View
            style={
              Object {
                "flexGrow": 0,
                "flexShrink": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "flexGrow": 1,
                }
              }
              x="2px 290fr 0px"
              y="0px minmax(0px, max-content) 2px"
            >
              <View
                style={
                  Object {
                    "flexGrow": 0,
                    "flexShrink": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexGrow": 1,
                    }
                  }
                  x="0px 290fr 0px"
                  y="0px minmax(0px, max-content) 0px"
                >
                  <View
                    style={
                      Object {
                        "flexGrow": 0,
                        "flexShrink": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flexGrow": 1,
                          "width": "100%",
                        }
                      }
                      x="0px 290fr 0px"
                      y="0px minmax(0px, max-content) 0px"
                    >
                      <View
                        absolute={true}
                        style={
                          Object {
                            "flexGrow": 1,
                          }
                        }
                        x="0px 288fr 2px"
                        y="0px minmax(0px, max-content) 153px"
                      >
                        <View
                          style={
                            Object {
                              "flexGrow": 0,
                              "flexShrink": 1,
                            }
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {},
                                Object {
                                  "flexGrow": 1,
                                  "height": 29,
                                  "marginBottom": 0,
                                  "marginLeft": 0,
                                  "marginRight": 0,
                                  "marginTop": 2,
                                },
                              ]
                            }
                          >
                            <View
                              accessibilityIgnoresInvertColors={true}
                              style={
                                Array [
                                  Object {
                                    "resizeMode": "contain",
                                  },
                                  Object {
                                    "bottom": -5,
                                    "height": 24,
                                    "left": 0,
                                    "position": "absolute",
                                    "width": 24,
                                  },
                                ]
                              }
                            >
                              <Image
                                source={
                                  Object {
                                    "testUri": "../../../libs/client/ui/src/lib/assets/79679bee68e57c649afa27a8de7f3414.png",
                                  }
                                }
                                style={
                                  Array [
                                    Object {
                                      "bottom": 0,
                                      "left": 0,
                                      "position": "absolute",
                                      "right": 0,
                                      "top": 0,
                                    },
                                    Object {
                                      "height": 24,
                                      "width": 24,
                                    },
                                    undefined,
                                  ]
                                }
                              />
                            </View>
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignItems": "flex-start",
                                    "flexDirection": "row",
                                    "justifyContent": "center",
                                  },
                                  Object {
                                    "bottom": -3,
                                    "left": 48,
                                    "position": "absolute",
                                    "width": 116,
                                  },
                                ]
                              }
                            >
                              <Text
                                ellipsizeMode="clip"
                                style={
                                  Object {
                                    "color": "#344053ff",
                                    "fontFamily": "System",
                                    "fontSize": 18,
                                    "fontStyle": "normal",
                                    "fontWeight": "400",
                                    "letterSpacing": 0,
                                    "lineHeight": 20,
                                    "paddingHorizontal": 0,
                                    "paddingVertical": 0,
                                    "textAlign": "center",
                                  }
                                }
                              >
                                Change email
                              </Text>
                            </View>
                          </View>
                        </View>
                        <View
                          style={
                            Object {
                              "flexGrow": 0,
                              "flexShrink": 1,
                            }
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "#d0d5ddff",
                                },
                                Object {
                                  "flexGrow": 1,
                                  "height": 1,
                                  "marginBottom": 1,
                                  "marginLeft": 0,
                                  "marginRight": 0,
                                  "marginTop": 19,
                                },
                              ]
                            }
                          />
                        </View>
                      </View>
                      <View
                        absolute={true}
                        style={
                          Object {
                            "flexGrow": 1,
                          }
                        }
                        x="0px 288fr 2px"
                        y="51px minmax(0px, max-content) 102px"
                      >
                        <View
                          style={
                            Object {
                              "flexGrow": 0,
                              "flexShrink": 1,
                            }
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {},
                                Object {
                                  "flexGrow": 1,
                                  "height": 29,
                                  "marginBottom": 0,
                                  "marginLeft": 0,
                                  "marginRight": 0,
                                  "marginTop": 2,
                                },
                              ]
                            }
                          >
                            <View
                              accessibilityIgnoresInvertColors={true}
                              style={
                                Array [
                                  Object {
                                    "resizeMode": "contain",
                                  },
                                  Object {
                                    "bottom": -5,
                                    "height": 24,
                                    "left": 0,
                                    "position": "absolute",
                                    "width": 24,
                                  },
                                ]
                              }
                            >
                              <Image
                                source={
                                  Object {
                                    "testUri": "../../../libs/client/ui/src/lib/assets/de8bda761eb9ceb32e01cf462e01ee7b.png",
                                  }
                                }
                                style={
                                  Array [
                                    Object {
                                      "bottom": 0,
                                      "left": 0,
                                      "position": "absolute",
                                      "right": 0,
                                      "top": 0,
                                    },
                                    Object {
                                      "height": 24,
                                      "width": 24,
                                    },
                                    undefined,
                                  ]
                                }
                              />
                            </View>
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignItems": "flex-start",
                                    "flexDirection": "row",
                                    "justifyContent": "center",
                                  },
                                  Object {
                                    "bottom": -3,
                                    "left": 50,
                                    "position": "absolute",
                                    "width": 150,
                                  },
                                ]
                              }
                            >
                              <Text
                                ellipsizeMode="clip"
                                style={
                                  Object {
                                    "color": "#344053ff",
                                    "fontFamily": "System",
                                    "fontSize": 18,
                                    "fontStyle": "normal",
                                    "fontWeight": "400",
                                    "letterSpacing": 0,
                                    "lineHeight": 20,
                                    "paddingHorizontal": 0,
                                    "paddingVertical": 0,
                                    "textAlign": "center",
                                  }
                                }
                              >
                                Change password
                              </Text>
                            </View>
                          </View>
                        </View>
                        <View
                          style={
                            Object {
                              "flexGrow": 0,
                              "flexShrink": 1,
                            }
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "#d0d5ddff",
                                },
                                Object {
                                  "flexGrow": 1,
                                  "height": 1,
                                  "marginBottom": 1,
                                  "marginLeft": 0,
                                  "marginRight": 0,
                                  "marginTop": 19,
                                },
                              ]
                            }
                          />
                        </View>
                      </View>
                      <View
                        absolute={true}
                        style={
                          Object {
                            "flexGrow": 1,
                          }
                        }
                        x="0px 288fr 2px"
                        y="102px minmax(0px, max-content) 51px"
                      >
                        <View
                          style={
                            Object {
                              "flexGrow": 0,
                              "flexShrink": 1,
                            }
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {},
                                Object {
                                  "flexGrow": 1,
                                  "height": 29,
                                  "marginBottom": 0,
                                  "marginLeft": 0,
                                  "marginRight": 0,
                                  "marginTop": 2,
                                },
                              ]
                            }
                          >
                            <View
                              accessibilityIgnoresInvertColors={true}
                              style={
                                Array [
                                  Object {
                                    "resizeMode": "contain",
                                  },
                                  Object {
                                    "bottom": -5,
                                    "height": 24,
                                    "left": 0,
                                    "position": "absolute",
                                    "width": 24,
                                  },
                                ]
                              }
                            >
                              <Image
                                source={
                                  Object {
                                    "testUri": "../../../libs/client/ui/src/lib/assets/c2a1ee46204a89d839eacd0921c2b2f7.png",
                                  }
                                }
                                style={
                                  Array [
                                    Object {
                                      "bottom": 0,
                                      "left": 0,
                                      "position": "absolute",
                                      "right": 0,
                                      "top": 0,
                                    },
                                    Object {
                                      "height": 24,
                                      "width": 24,
                                    },
                                    undefined,
                                  ]
                                }
                              />
                            </View>
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignItems": "flex-start",
                                    "flexDirection": "row",
                                    "justifyContent": "center",
                                  },
                                  Object {
                                    "bottom": -3,
                                    "left": 49,
                                    "position": "absolute",
                                    "width": 122,
                                  },
                                ]
                              }
                            >
                              <Text
                                ellipsizeMode="clip"
                                style={
                                  Object {
                                    "color": "#344053ff",
                                    "fontFamily": "System",
                                    "fontSize": 18,
                                    "fontStyle": "normal",
                                    "fontWeight": "400",
                                    "letterSpacing": 0,
                                    "lineHeight": 20,
                                    "paddingHorizontal": 0,
                                    "paddingVertical": 0,
                                    "textAlign": "center",
                                  }
                                }
                              >
                                Change colour
                              </Text>
                            </View>
                          </View>
                        </View>
                        <View
                          style={
                            Object {
                              "flexGrow": 0,
                              "flexShrink": 1,
                            }
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "#d0d5ddff",
                                },
                                Object {
                                  "flexGrow": 1,
                                  "height": 1,
                                  "marginBottom": 1,
                                  "marginLeft": 0,
                                  "marginRight": 0,
                                  "marginTop": 19,
                                },
                              ]
                            }
                          />
                        </View>
                      </View>
                      <View
                        style={
                          Object {
                            "flexGrow": 1,
                            "width": "100%",
                          }
                        }
                        x="0px 290fr 0px"
                        y="153px minmax(0px, max-content) 0px"
                      >
                        <View
                          style={
                            Object {
                              "flexDirection": "row",
                              "flexGrow": 1,
                            }
                          }
                          x="2px 288fr 0px"
                          y="14px minmax(0px, max-content) 13px"
                        >
                          <View
                            style={
                              Object {
                                "flexBasis": 24,
                                "flexGrow": 0,
                                "flexShrink": 1,
                              }
                            }
                          >
                            <View
                              accessibilityIgnoresInvertColors={true}
                              style={
                                Array [
                                  Object {
                                    "resizeMode": "contain",
                                  },
                                  Object {
                                    "height": 24,
                                    "marginBottom": 0,
                                    "marginLeft": 0,
                                    "marginRight": 0,
                                    "marginTop": 1,
                                    "minWidth": 24,
                                    "width": 24,
                                  },
                                ]
                              }
                            >
                              <Image
                                source={
                                  Object {
                                    "testUri": "../../../libs/client/ui/src/lib/assets/754d89ce51e0854b878e2ef557f04bde.png",
                                  }
                                }
                                style={
                                  Array [
                                    Object {
                                      "bottom": 0,
                                      "left": 0,
                                      "position": "absolute",
                                      "right": 0,
                                      "top": 0,
                                    },
                                    Object {
                                      "height": 24,
                                      "width": 24,
                                    },
                                    undefined,
                                  ]
                                }
                              />
                            </View>
                          </View>
                          <View
                            style={
                              Object {
                                "flexBasis": 24,
                                "flexGrow": 0,
                                "flexShrink": 1,
                              }
                            }
                          />
                          <View
                            style={
                              Object {
                                "flexGrow": 0,
                                "flexShrink": 0,
                                "minWidth": 0,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "alignItems": "flex-start",
                                  "flexDirection": "row",
                                  "flexGrow": 1,
                                  "justifyContent": "center",
                                }
                              }
                              x="0px 65fr 175px"
                              y="3px minmax(0px, max-content) 2px"
                            >
                              <Text
                                ellipsizeMode="clip"
                                style={
                                  Object {
                                    "color": "#344053ff",
                                    "fontFamily": "System",
                                    "fontSize": 18,
                                    "fontStyle": "normal",
                                    "fontWeight": "400",
                                    "letterSpacing": 0,
                                    "lineHeight": 20,
                                    "paddingHorizontal": 0,
                                    "paddingVertical": 0,
                                    "textAlign": "center",
                                  }
                                }
                              >
                                Log out
                              </Text>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexGrow": 0,
          "flexShrink": 1,
          "opacity": 1,
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Array [
            Object {
              "resizeMode": "contain",
            },
            Object {
              "height": 18,
              "marginBottom": 0,
              "marginLeft": "auto",
              "marginRight": "auto",
              "marginTop": 396,
              "minWidth": 9,
              "width": 9,
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../libs/client/ui/src/lib/assets/f762e82c40c63fd888499d8d139bcf08.png",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 18,
                "width": 9,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;
